quality_rules_ecalc.php

```diff
@@ -0,0 +1,273 @@
+	{
+?>
+     <option value="<?php echo $module->escapeHTML( $groupID ); ?>">
+      <?php echo $module->escapeHTML( $groupName ), "\n"; ?>
+     </option>
+<?php
+	}
+}
+?>
+   </select>
+  </td>
+ </tr>
+ <tr>
+  <td><?php echo $GLOBALS['lang']['dataqueries_292']; ?></td>
+  <td style="padding-left:5px;padding-bottom:5px">
+   <input id="fixcalcs" type="checkbox">
+  </td>
+ </tr>
+ <tr>
+  <td></td>
+  <td><input id="exec" type="button" value="<?php echo $GLOBALS['lang']['dataqueries_80']; ?>"></td>
+ </tr>
+</table>
+<div id="progress" style="display:none">
+ <?php echo $GLOBALS['lang']['data_import_tool_338']; ?> <span id="progress1"></span> /
+ <span id="progress2"></span>
+ <br>
+ <progress id="progressbar" style="min-width:60%"></progress>
+</div>
+<table id="resultstable" style="display:none;width:97%;margin-top:10px">
+ <thead>
+  <th style="width:150px"><?php echo $GLOBALS['lang']['global_49']; ?></th>
+  <th><?php echo $GLOBALS['lang']['dataqueries_23']; ?></th>
+ </thead>
+ <tbody>
+ </tbody>
+</table>
+<script type="text/javascript">
+$('#resultstable th').css('font-weight', 'bold').css('padding', '5px')
+$('#resultstable th').css('border', '1px solid #ccc').css('background', '#ececec')
+$('#exec').on('click',function()
+{
+  $('#exec').prop( 'disabled', true )
+  var vFixCalcs = $('#fixcalcs').prop('checked')
+  $.ajax( { url : '<?php echo $module->getUrl( 'quality_rules_ecalc.php' ); ?>',
+            method : 'POST',
+            data : { arm : $('#armselect').val(), dag : $('#dagselect').val() },
+            headers : { 'X-RC-UITweak-DQR' : 'records' },
+            dataType : 'json'
+          } )
+   .done( function( vRecordList )
+   {
+     $('#progress1').text('0')
+     $('#progress2').text( vRecordList.length )
+     $('#progressbar').attr( 'max', vRecordList.length+1 ).attr( 'value', '0' )
+     $('#progress').css('display', '')
+     $('#resultstable').css('display', '')
+     $('#resultstable tbody').html('')
+     var vEven = false
+     var vFnExec = function()
+     {
+       if ( vRecordList.length == 0 )
+       {
+         $('#progressbar').attr( 'value', $('#progressbar').attr('max') )
+         setTimeout( function(){ $('#progress').css('display','none') }, 3000 )
+         $('#exec').prop('disabled', false)
+         return
+       }
+       $('#progressbar').attr( 'value', $('#progressbar').attr('max')-vRecordList.length )
+       $('#progress1').text( $('#progressbar').attr('max')-vRecordList.length )
+       var vRecordID = vRecordList.shift()
+       $.ajax( { url : '<?php echo $module->getUrl( 'quality_rules_ecalc.php' ); ?>',
+                 method : 'POST',
+                 data : { action : vFixCalcs ? 'fixCalcs' : '',
+                          record : vRecordID },
+                 headers : { 'X-RC-UITweak-DQR' : 'calcs' },
+                 dataType : 'json'
+               } )
+        .done( function( vResponse )
+        {
+          var vTdRecord = $('<td></td>').text( vResponse.record )
```

Comment:
If 'Fix ALL incorrect calculated values' is not checked it would be nice to have a fix on individual records so you don't need to start again.